appcues/src/main/java/com/appcues/ui/primitive/OptionSelectPrimitive.kt

```diff
@@ -0,0 +1,139 @@
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.material.DropdownMenu
+import androidx.compose.material.DropdownMenuItem
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.zIndex
+import com.appcues.R
+import com.appcues.data.model.ExperiencePrimitive.OptionSelectPrimitive
+import com.appcues.data.model.styling.ComponentControlPosition
+import com.appcues.data.model.styling.ComponentControlPosition.BOTTOM
+import com.appcues.data.model.styling.ComponentControlPosition.HIDDEN
+import com.appcues.data.model.styling.ComponentControlPosition.LEADING
+import com.appcues.data.model.styling.ComponentControlPosition.TOP
+import com.appcues.data.model.styling.ComponentControlPosition.TRAILING
+import com.appcues.data.model.styling.ComponentDisplayFormat.HORIZONTAL_LIST
+import com.appcues.data.model.styling.ComponentDisplayFormat.PICKER
+import com.appcues.data.model.styling.ComponentDisplayFormat.VERTICAL_LIST
+import com.appcues.data.model.styling.ComponentSelectMode
+import com.appcues.data.model.styling.ComponentSelectMode.MULTIPLE
+import com.appcues.data.model.styling.ComponentSelectMode.SINGLE
+import com.appcues.ui.extensions.getHorizontalAlignment
+
+@Composable
+internal fun OptionSelectPrimitive.Compose(modifier: Modifier) {
+    Column(
+        modifier = modifier,
+        horizontalAlignment = style.getHorizontalAlignment(),
+    ) {
+        label.Compose()
+
+        when {
+            selectMode == SINGLE && displayFormat == PICKER -> {
+                var expanded by remember { mutableStateOf(false) }
+                var selectedIndex by remember { mutableStateOf(0) }
+
+                Box(modifier = Modifier.clickable(onClick = { expanded = true })) {
+                    options[selectedIndex].content.Compose()
+                    DropdownMenu(
+                        expanded = expanded,
+                        onDismissRequest = { expanded = false },
+                        modifier = Modifier
+                            .background(Color.White)
+                    ) {
+                        options.forEachIndexed { index, optionItem ->
+                            DropdownMenuItem(onClick = {
+                                selectedIndex = index
+                                expanded = false
+                            }) {
+                                optionItem.content.Compose()
+                            }
+                        }
+                    }
```

Comment:
using a `DropdownMenu` as an initial, very basic way to do a Picker.

For reference, a default SwiftUI Picker on iOS looks like:
![ios_picker](https://user-images.githubusercontent.com/19266448/187492080-a4ff20d6-afe4-4e8e-8c17-131e130ec3e6.gif)

Reply:
thats how I imagine we would do on Android so you got that right